cluster/core/src/main/scala/com/lightbend/lagom/internal/cluster/JoinClusterImpl.scala

```diff
@@ -31,8 +35,21 @@ private[lagom] object JoinClusterImpl {
         "conflict with Akka Cluster Bootstrap or cause split-brain clusters.")
     }
 
-    if (cluster.settings.SeedNodes.isEmpty && joinSelf)
+    if (clusterBootstrapEnabled && joinSelf) {
+      throw new IllegalArgumentException(
+        "Both \"lagom.cluster.bootstrap.enabled\" and \"lagom.cluster.join-self\" are enabled, you should enable only one. " +
+          "Typically, \"lagom.cluster.bootstrap.enabled\" should be used in production while \"lagom.cluster.join-self\" in development and test environments"
+      )
+    }
+
+    // we give priority
+    if (clusterBootstrapEnabled) {
+      // TODO: move AkkaManagement to Guice module
```

Comment:
Is this a lot of work or is it something we can solve here at this PR?

Reply:
I prefer to make it on another PR. Eventually in 1.5.1. That's not so critical.